### ISSUE TITLE
include that ability to pass an already created figure as a parameter to the many plot methods

### ISSUE THREAD

**Issue by daliagachc** [pydata/xarray#7401]
### Is your feature request related to a problem?

Sometime i need to combine xarray Facet grids with other plots. It would be amazing if I could pass a created figure to the function. Event better a subfigure so that the possibilities are infinite! 

### Describe the solution you'd like

for example: 
```python
  da = xr.tutorial.open_dataset("air_temperature")['air']
  f = plt.figure()
  (
      da
      [{'time':[1,2,4]}]
      .plot.contourf(col='time',fig = f)
  )

```

### Describe alternatives you've considered

an alternative is to manually to all plots in a created figure, but this becomes cumbersome.

**Comment by mathause** [comment]
I think the usual workflow is to create the figure with gridspec and then update the created subplots and figure. Is there something that is not possible if done this way around?

**Comment by daliagachc** [comment]
thanks for the reply. Somehow i don't find an elegant way that does not involve many loops and hoops. Lets consider  the following short example:
I want to produce a figure that takes the xr.tutorial.open_dataset("air_temperature") and produces a mean grouped by season and hour in the same figure: 
<img width="876" alt="image" src="https://user-images.githubusercontent.com/15239248/209472922-1e5e25a1-5a7a-4e4a-ae51-d90d7a14dddc.png">
Creating the top and bottom rows independently is super efficient with xarray: 
```python 
import xarray as xr
import matplotlib.pyplot as plt

da = xr.tutorial.open_dataset("air_temperature")['air']

(
    da.groupby(da['time'].dt.season)
    .mean()
    .plot(col='season')
)

(
    da.groupby(da['time'].dt.hour)
    .mean()
    .plot(col='hour')
)
```
<img width="491" alt="image" src="https://user-images.githubusercontent.com/15239248/209473161-eca3b655-e1b8-4a4a-bd42-8e60f559a23d.png">

but combining both seems quite complicated to me. I wish I could pass a subfigure to the plot function in the following way 
```python
fig:plt.Figure = plt.figure()
(fig_top,fig_bot) = fig.subfigures(2,1)

(
    da.groupby(da['time'].dt.season)
    .mean()
    .plot(col='season',fig=fig_top)
)

(
    da.groupby(da['time'].dt.hour)
    .mean()
    .plot(col='hour', fig=fig_bot)
)
```
![image](https://user-images.githubusercontent.com/15239248/209473217-49b5f198-4bca-41dc-b955-3ad1c3992626.png)

maybe I am missing a smart way of doing this? 
cheers

**Comment by daliagachc** [comment]
I have actually manage to make it work with minimal modification of the source xarray code:
```python 
import xarray as xr
import matplotlib.pyplot as plt
import matplotlib as mpl


def _pass(*_,**__):
    pass

mpl.figure.SubFigure.tight_layout = _pass

da = xr.tutorial.open_dataset("air_temperature")['air']
fig:plt.Figure = plt.figure(
    constrained_layout=True,
    figsize = (2*4,2*2)
)

(fig_top,fig_bot) = fig.subfigures(2,1)

(
    da.groupby(da['time'].dt.season)
    .mean()
    .plot(col='season',fig=fig_top)
)

(
    da.groupby(da['time'].dt.hour)
    .mean()
    .plot(col='hour', fig=fig_bot)
)
```
![image](https://user-images.githubusercontent.com/15239248/209475744-e210de9d-a336-4ef2-969b-5130c9811505.png)

**Comment by mathause** [comment]
Good point. Combining two `FacetGrids` is a good use case. My intuition would be to allow passing _exactly_ the correct number of axes and not the figure. But then I have not heard from subfigure before now and this may also be a valid option.

```python
f, axs = plt.subplots(2, 4)
da1.plot(col='season', ax=axs[0, :])
da2.plot(col='season', ax=axs[1, :])
```
